modules/custom-status/lib/custom-status-block.js

```diff
@@ -190,11 +197,31 @@ const getCustomSaveButtonText = ( nextStatusTerm, isWideViewport ) => {
 			// translators: %s: Next custom status name, possibly truncated with an ellipsis. e.g. "Draft" or "Pendi…"
 			buttonText = sprintf( __( 'Move to %s', 'vip-workflow' ), truncatedStatus );
 		}
+	} else if ( ! nextStatusTerm && isRestrictedStatus ) {
+		// Awaiting a privileged user to approve publishing.
+		// Show disabled "Publish" button as a placeholder.
+		buttonText = __( 'Publish', 'vip-workflow' );
 	}
 
 	return buttonText;
 };
 
+const isStatusRestrictedFromMovement = status => {
```

Comment:
I feel like with the switch to the current_user, the only piece that would need some guarding would be the `transition_post_status`. 

But, if are there too many bugs one thing we could do:

- Take away the edit_post capability so no save operation would work for the user that authored the post as long as the status is restricted.
- They can move the post back a status to edit it's content, and then send it back to the current status.

This would change the paradigm a touch but just a backup plan in case we need it.

Reply:
> - Take away the `edit_post` capability so no save operation would work for the user that authored the post as long as the status is restricted.

This is what I was thinking, as this effectively stopped transitions in the UI in our prior work. The main problem was the UI and weird error message when that failed. However, if we have the UI covered and this is only a backup solution when the client-side code is manipulated, I think it'll work well.

> - They can move the post back a status to edit it's content, and then send it back to the current status.

As of right now a user can still "Save draft" on a restricted status. This makes sense to me if we're in a "Pending Review"-type stage where only the editor can approve, but an author may be asked to make changes to the content in that stage. So the content is always editable, but the status should not be.